crates/store/src/db/sql.rs

```diff
@@ -665,6 +665,58 @@ pub fn select_notes_by_id(conn: &mut Connection, note_ids: &[NoteId]) -> Result<
     Ok(notes)
 }
 
+/// Select note inclusion proofs matching the NoteId, using the given [Connection].
+///
+/// # Returns
+///
+/// - Empty map if no matching `note`.
+/// - Otherwise, note inclusion proofs, which `note_id` matches the `NoteId` as bytes.
+pub fn select_block_note_inclusion_proofs(
+    conn: &mut Connection,
+    note_ids: &[NoteId],
+) -> Result<BTreeMap<NoteId, NoteInclusionProof>> {
+    let note_ids: Vec<Value> = note_ids.iter().map(|id| id.to_bytes().into()).collect();
+
+    let mut select_notes_stmt = conn.prepare(
+        "
+        SELECT
+            block_num,
+            note_id,
+            batch_index,
+            note_index,
+            merkle_path
+        FROM
+            notes
+        WHERE
+            note_id IN rarray(?1)
+        ORDER BY
+            block_num ASC
+        ",
+    )?;
+
+    let mut result = BTreeMap::new();
+    let mut rows = select_notes_stmt.query(params![Rc::new(note_ids)])?;
+    while let Some(row) = rows.next()? {
+        let block_num = row.get(0)?;
+
+        let note_id_data = row.get_ref(1)?.as_blob()?;
+        let note_id = NoteId::read_from_bytes(note_id_data)?;
+
+        let batch_index = row.get(2)?;
+        let note_index = row.get(3)?;
+        let node_index_in_block = BlockNoteIndex::new(batch_index, note_index).to_absolute_index();
+
+        let merkle_path_data = row.get_ref(4)?.as_blob()?;
+        let merkle_path = MerklePath::read_from_bytes(merkle_path_data)?;
```

Comment:
Meta: something to consider is adding typed getters using an extension trait for the sqlite row. We did something similar in pathfinder [here](https://github.com/eqlabs/pathfinder/blob/7ec71dcbc35372a4424f4772b3a02ecfe7e19c5c/crates/storage/src/params.rs#L151) which I think added quite a lot in terms of consistency and readability.

Reply:
Created #442